packages/rancher-monitoring/generated-changes/overlay/templates/grafana/namespaces.yaml

```diff
@@ -1,10 +1,12 @@
 {{- if and .Values.grafana.enabled }}
 {{- if or .Values.grafana.sidecar.dashboards.enabled .Values.grafana.defaultDashboardsEnabled }}
+{{- if or (not .Values.grafana.sidecar.dashboards.useExistingNamespace) (eq .Values.grafana.sidecar.dashboards.searchNamespace "ALL") }}
```

Comment:
There are 3 possible inputs that we expect:
1. searchNamespace = ALL: ignore useExistingNamespace and create the cattle-dashboards namespace by default
2. searchNamespace != ALL && useExistingNamespace=true: don't create a namespace, just create resources in an existing namespace
3. searchNamespace != ALL && useExistingNamespace=false: create the namespace and create resources within it.

Question: Should we also split case 1 based on useExistingNamespace?

The only case in which this would matter is if you already created the cattle-dashboard namespace from another chart / in a way that is not managed by Helm and you want to use that as your existing namespace. But I don't see that as a likely possibility.

vs. it's extremely likely a user would have left useExistingNamespace=true when switching searchNamespace = ALL, which would result in a helm error since the namespace we are creating resources within (default cattle-dashboard) no longer exists.

Reply:
Based on offline discussion with @brendarearden, I'll be making changes to this PR to logically separate searchNamespace from defaultDashboardsNamespace, which will alleviate the problem I brought up here. Will resolve this comment and re-request both reviews once that change is tested / in.

Reply:
Changes have been made.